src/Mobs/PathFinder.cpp

```diff
@@ -0,0 +1,233 @@
+
+
+#include "Globals.h"
+#include "PathFinder.h"
+#include "../Chunk.h"
+
+cPathFinder::cPathFinder(double a_MobWidth, double a_MobHeight)
+	: m_Path(nullptr)
+	, m_GiveUpCounter(0)
+	, m_TicksSinceLastPathReset(1000)
```

Comment:
>I suppose it comes down to a matter of performance versus accuracy.

It is, up to a certain point. 

Beyond that point, you're just killing performance without gaining noticable accuracy. For instance, there is very little visible difference between the current ticking method and your proposed 1-block recalc method. But there is a huge performance difference. Also, once I make the mobs ditch the pathfinder and just move in a straight line towards target when there is a clear line of sight to target, you'd have to be really sharp eyed to see any difference at all. So if we go the 1-block-recalc route, we trade performance away and practically gain nothing.

>For selective recalculation, you'll definitely encounter terrain layouts where even a one block change in destination could mean a completely different path, and so a player would see a monster walking towards them for X ticks, stop, and turn back to head along a different route

Ah, yes. I like to call it the "meta-stable-spot problem": The mob is at a special point where an inch of player movement changes the shortest path completely. That problem is present in both methods of recalculation in different forms. In some cases the delayed recalc is actually an advantage here: The mob just moves along one path (and doesn't recalc), making it leave the meta-stable-spot faster. In other terrain cases 1-block-recalc may behave better. You can't make a case in favor of 1-block-recalc here. I think the problem is too rare and too harmless to be worth taking into account anyways.

I see no single rational reason in favor of 1-block-recalc. If you do have one, please enlighten me.

Why not to use 1-block-recalc:
 - It calls cPath every single tick. That's much slower and not justifiable. 
 - We already have a working code for the tick+distance method.
 - The accuracy advantage is barely noticbale, will be virtually unnoticable when line-of-sight optimization is done.
 - That meta-stable spot issue is present in both methods.
 - Why would a mob which is 40-block away from a player recalculate every inch of player movement? That doesn't make sense. 
 - The current algorithm increases the recalc frequency significantly as the player moves closer, up to a certain max frequency, we can increase that max frequency if needed. If necessary, we could  remove the max frequency restriction altogether, making it calculate as frequent as 1-block-recalc on close range, and less often on long range. But even that much accuracy is unneeded.